python/revng/scripting/daemon_project.py

```diff
@@ -0,0 +1,216 @@
+
+        query = gql(
+            """
+            query($step: String!, $paths: String!, $index: BigInt!) {
+                produceArtifacts(step: $step, paths: $paths, index: $index) {
+                    ... on Produced {
+                        result
+                    }
+                }
+            }
+        """
+        )
+        variables = {
+            "step": artifact_name,
+            "paths": ",".join(targets),
+            "index": self._get_content_commit_index(),
+        }
+
+        response = self._make_request(query, variables)
+        return response["produceArtifacts"]["result"].encode()
+
+    def fetch_many(self, params: dict[str, list[Any]]) -> dict:
+        result = {}
+        for artifact_name, targets in params.items():
+            result[artifact_name] = self.fetch(artifact_name, targets)
+
+        return result
+
+    def analyze(self, analysis_name: str, target_class: list[Any] = []) -> Binary:
+        container_input, container_kind = self._get_analysis_inputs(analysis_name)
+        step = self._get_step_name(analysis_name)
+
+        variables = {
+            "step": step,
+            "analysis": analysis_name,
+            "index": self._get_content_commit_index(),
+        }
+        if not container_input:
+            query = gql(
+                """
+                mutation($step: String!, $analysis: String!, $index: BigInt!) {
+                    runAnalysis(step: $step, analysis: $analysis, index: $index) {
+                        __typename
+                    }
+                }
+            """
+            )
+        else:
+            query = gql(
+                """
+                mutation($step: String!, $analysis: String!, $container: String!, $index: BigInt!) {
+                    runAnalysis(step: $step, analysis: $analysis,
+                                containerToTargets: $container, index: $index) {
+                        __typename
+                    }
+                }
+            """
+            )
+            targets = []
+            if container_kind != "root":
+                if target_class:
+                    for target in target_class:
+                        if type(target).__name__ == "Function":
+                            targets.append(f"{target.Entry}:{container_kind}")
+                        else:
+                            raise RuntimeError(f"Target {type(target).__name__} not implemented")
+                else:
+                    for function in self.model.Functions:
+                        targets.append(f"{function.Entry}:{container_kind}")
+            else:
+                targets = [f":{container_kind}"]
+            variables["container"] = json.dumps({container_input: targets})
+
+        self._make_request(query, variables)
+        model = self._get_model()
+        self.set_model(model)
+        return model
+
+    def analyses_list(self, analysis_name: str):
```

Comment:
This seems unused

Reply:
This runs the query `runAnalysesList` and it's used in the [test](https://github.com/revng/revng/blob/138d75d6d60d9d91705fbf77ac2344c0d7885a8b/share/revng/test/configuration/revng/scripting.yml#L119)

Reply:
Ok, but then this should be duplicated in `CLIProject`

Reply:
I made `analyses_list` a private method and it's called by `analyze` when the parameter matches the name of an analysis list, this way the interface for the `analyze` is the same for `cli` and `daemon`.

Reply:
I'd expose it for both Project classes, the analyses lists offer a simplified way of calling analyses compared to the single analyses (where the container mapping needs to be passed)